Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -825,6 +827,11 @@ LayoutUnit GridTrackSizingAlgorithmStrategy::logicalHeightForChild(RenderBox& ch
     if (shouldClearOverridingContainingBlockContentSizeForChild(child, ForRows)) {
         setOverridingContainingBlockContentSizeForChild(*renderGrid(), child, childBlockDirection, std::nullopt);
         child.setNeedsLayout(MarkOnlyThis);
+
+        if (is<RenderBox>(child.firstChild())) {
+            for (auto itr = RenderIterator<RenderBox>(&child, child.firstChildBox()); itr; itr = itr.traverseNext())
+                itr->setNeedsLayout(MarkOnlyThis);
```

Comment:
Why is it OK for a `const` getter called `logicalHeightForChild` to set children as needing layout?

Reply:
@smfr: I'm not sure either, but note that existing code already does this, so perhaps this is expected here.

Reply:
@stwrt Do we not have a visitor thing that knows how to walk through the children and mark them as needing layout? I vaguely remember this existing, but I could be wrong.

Reply:
I think we might want something like the following (though it looks like other render code uses your approach, so it might be preferable):

```
 for (auto& child : descendantsOfType<RenderBox>(child.firstChild()))
         child.setNeedsLayout(MarkOnlyThis);
```

Reply:
Also this fix looks very specific to the case where the nested grid is a second level child of the grid item. I assume dirty bits need to be propagated further down in case of deeper block level structure.

Reply:
> I think we might want something like the following (though it looks like other render code uses your approach, so it might be preferable):
> 
> ```
>  for (auto& child : descendantsOfType<RenderBox>(child.firstChild()))
>          child.setNeedsLayout(MarkOnlyThis);
> ```
descendant iterator may dirty more that what we need (as it traverses the entire subtree). The issue here is that marking only the direct children (grid items) is not sufficient but the same time marking the entire subtree may be an overkill. In theory such invalidation should stop at formatting context boundary, but we don't have such functionality in render tree yet.

Reply:
I started a debugging session on this to be able to provide a (semi)educated review feedback..but it should not stop others from reviewing it (and landing) if urgent.